common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -440,7 +440,7 @@ protected StringBuilder toStringBuilder() {
      * @return The executor used to notify listeners when this promise is complete.
      */
     @Override
-    public final EventExecutor executor() {
+    public EventExecutor executor() {
```

Comment:
@ejona86 I will need to have a look. That said this has nothing todo with this PR and so I would like to handle this as a followup.

Reply:
It isn't unrelated because the semantics are changing. I'd totally believe some code would (properly) be assuming this return value never changes. Because it is public the "blast radius" is harder to determine, not just classes in the Promise hierarchy. We sort of need to audit all callers to make sure they are safe with this change.

Reply:
fixed this...